lib/node_modules/@stdlib/array/complex64/docs/types/index.d.ts

```diff
@@ -677,6 +677,42 @@ declare class Complex64Array implements Complex64ArrayInterface {
 	*/
 	lastIndexOf( searchElement: ComplexLike, fromIndex?: number ): number;
 
+	/**
+	* Returns a new array with each element being the result of the callback function.
+	*
+	* @param fcn - callback function
+	* @param thisArg - execution context
+	* @returns new array containing elements which are the result of the callback function
+	*
+	* @example
+	* var Complex64 = require( '@stdlib/complex/float32' );
+	* var realf = require( '@stdlib/complex/realf' );
+	* var imagf = require( '@stdlib/complex/imagf' );
+	*
+	* function scale( v, i ) {
+	*     return new Complex64( 2.0*realf( v ), 2.0*imagf( v ) );
+	* }
+	*
+	* var arr = new Complex64Array( 3 );
+	*
+	* arr.set( [ 1.0, -1.0 ], 0 );
+	* arr.set( [ 2.0, -2.0 ], 1 );
+	* arr.set( [ 3.0, -3.0 ], 2 );
+	*
+	* var out = arr.map( scale );
+	* // returns <Complex64Array>
+	*
+	* var z = out.get( 0 );
+	* // returns <Complex64>
+	*
+	* var re = realf( z );
+	* // returns 2.0
+	*
+	* var im = imagf( z );
+	* // returns -2.0
+	*/
+	map<U = unknown>( fcn: Callback<U>, thisArg?: ThisParameterType<Callback<U>> ): Complex64Array;
```

Comment:
@Jaysukh-409 You can't use `Callback` as currently defined, as `Callback` is defined to return `void`. This matches what is needed for `forEach`, but not here. You'll want to create separate type definitions similar to `Callback` above, but supporting returning either a `ComplexLike` object or a two element array containing real and imaginary components. E.g.,

```
type NullaryMapFcn = ( this: U ) => ComplexLike | [ number, number ];

...

type MapFcn<U> = NullaryMapFcn<U> | UnaryMapFcn<U> | BinaryMapFcn<U> | TernaryMapFcn<U>;
```